ta/os_test/ta_arm_pauth.c

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, Linaro Limited
+ * All rights reserved.
+ */
+
+#include <inttypes.h>
+#include <string.h>
+#include <ta_os_test.h>
+#include <tee_internal_api.h>
+
+void corrupt_pac(void);
+
+/* Assuming VA_MASK considering maximum 48 bit of VA */
+#define VA_MASK		0xfffff0000000000
+
+TEE_Result ta_entry_pauth_test_nop(void)
+{
+	uint64_t pac = 0;
+	uint64_t ptr = 0;
+	bool implemented = false;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	res = TEE_GetPropertyAsBool(
+			TEE_PROPSET_TEE_IMPLEMENTATION,
+			"org.trustedfirmware.optee.cpu.feat_pauth_implemented",
+			&implemented);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	if (!implemented)
+		return TEE_ERROR_NOT_SUPPORTED;
+
+	/* Check if PAC instruction generates the authentication code */
+	for (int i = 0; i < 10; i++) {
+		asm volatile("paciza %0 " : "+r" (ptr));
```

Comment:
Please avoid inline assembly mixed with the code. I guess there's not much generic use of this instruction so putting at some common place might not be needed, but perhaps in a separate function above.
By the way, what is this trying to do? Just reading the description of the instruction in the reference manual still leaves some questions.

Reply:
Pointer authentication requires an address, modifier and key.
paczia Rd
With paczia the address is in the register Rd, modifier is 0 and key is APIAKEY
The instruction will generate and insert pointer authentication code on address specified in Rd with  modifier 0 and key A. The PAC is inserted in Rd itself.